src/control.py

```diff
@@ -1,35 +1,96 @@
-#Python script to read analog voltage values from ADS1115 16bit ADC with PGA
-#Author: Carson Berry 
-#Date: February 2nd, 2021
-#usage: 
-#    python3
-#    import src.adc as adc
-#    sensors = adc.adc_sensors()
-#    sensors.read_leak()
-#    sensors.read_pH()
-# Reference provided at: https://learn.adafruit.com/adafruit-4-channel-adc-breakouts/python-circuitpython
+
+#----------------------------------------------Template--------------------------------------------------#
+# This is a brief template on creating *new threading modules*
+# 1. Create a class, for example: class ph_control(object):
+# 2. Copy all the exiting initialization functions for sensors
+# 3. Add the new initialization function for the new sensor/device
+# 4. Set up the threading module at the end of this file, this includes:
+# 4.1. Define your class, for example: VARIABLE_NAME = NAME_OF_THE_CLASS()
+# 4.2. Create the thread module, for example: NAME_OF_THREAD = Thread(target = VARIABLE_NAME.FUNCTION_NAME) 
+# Notice that the FUNCTION_NAME should be the function in the class (defined before), and is the one that you would like it to iterate
+# 4.3. Start the thread, for example: NAME_OF_THREAD.start()
+
+
 
 
-import RPi.GPIO as GPIO
-import time
 import board
-import threading
 import busio
 import adafruit_ads1x15.ads1115 as ADS
 from adafruit_ads1x15.analog_in import AnalogIn
+from threading import Thread
+import RPi.GPIO as GPIO
+import time
+import src.pins
+
+
+
+class ph_control(object):
+    def __init__(self):
+        self.ads=0
+        self.init_i2c()
+        self.leak_sensor = 0
```

Comment:
Does the ph_control() object need to have the leak sensor defined? Can this be removed?

Reply:
Yes, I believe this can be removed. I will remove it for now and comment on the code what's removed, since the threading sometimes getting wired in init functions.